src/main/java/org/jenkinsci/plugins/workflow/cps/CpsFlowExecution.java

```diff
@@ -255,6 +255,18 @@
 
     private final AtomicInteger iota = new AtomicInteger();
 
+    /** Number of node IDs to use in lookup table, 500 covers most common flow graphs  */
+    private static final int ID_LOOKUP_TABLE_SIZE = 500;
+
+    /** Preallocated lookup table for small ID values, used instead of interning for speed & simplicity */
```

Comment:
Probably overkill. A direct call to `String.intern` is going to be so much faster than all the crap we are doing when creating a new `FlowNode` that using a lookup table is just gratuitous code.